src/terminal.go

```diff
@@ -1413,6 +1415,12 @@ func (t *Terminal) Loop() {
 			case actPageDown:
 				t.vmove(-(t.maxItems() - 1))
 				req(reqList)
+			case actHalfPageUp:
+				t.vmove(t.maxItems()/2 - 1)
+				req(reqList)
+			case actHalfPageDown:
+				t.vmove(-(t.maxItems()/2 - 1))
```

Comment:
`- 1` doesn't seem necessary here. The point of `t.maxItems() - 1` is to keep the scroll offset unchanged when you press pgup for the first time. It doesn't apply for half page movements.